backend/schedulerd/check_watcher.go

```diff
@@ -111,6 +111,10 @@ func (c *CheckWatcher) startWatcher() {
 	for {
 		select {
 		case watchEvent, ok := <-watchChan:
+			if watchEvent.CheckConfig == nil {
+				logger.Error("nil check config received from check config watcher")
+				return
+			}
```

Comment:
This is probably not quite right, as it doesn't check the `ok` value before accessing `watchEvent.CheckConfig`

Reply:
Good catch. Fixed!

Reply:
That's not quite right either, I think we need to stop the schedulers if this happens, like when `<-ctx.Done()`. :thinking:

Reply:
Maybe put the stopping logic in a defer.